processor/src/operations/horner_ops.rs

```diff
@@ -0,0 +1,502 @@
+    ///    evaluation procedure. (acc0', acc1') is the updated value of this accumulator.
+    /// 3. alpha_addr is the memory address of the evaluation point i.e., alpha.
+    ///
+    /// The instruction also makes use of the helper registers to hold the value of
+    /// alpha = (alpha0, alpha1) during the course of its execution.
+    pub(super) fn op_horner_eval_ext(&mut self) -> Result<(), ExecutionError> {
+        // --- read the values of the coefficients, over the extension field, from the stack ------
+        let coef = self.get_coeff_as_quad_ext_elements();
+
+        // --- read the evaluation point alpha from memory ----------------------------------------
+        let alpha = self.get_evaluation_point()?;
+
+        // --- read the accumulator values from stack ---------------------------------------------
+        let acc_old = self.get_accumulator();
+
+        // --- compute the updated accumulator values ---------------------------------------------
+        let acc_new = coef.iter().rev().fold(acc_old, |acc, coef| *coef + alpha * acc);
+
+        // --- copy the stack ---------------------------------------------------------
+        self.stack.copy_state(0);
+
+        // --- update the accumulators ------------------------------------------------------------
+        self.stack.set(ACC_HIGH_INDEX, acc_new.to_base_elements()[1]);
+        self.stack.set(ACC_LOW_INDEX, acc_new.to_base_elements()[0]);
+
+        // --- set the helper registers -----------------------------------------------------------
+        self.populate_helper_registers(alpha);
+
+        Ok(())
+    }
+
+    //// HELPER METHODS
+    //// ------------------------------------------------------------------------------------------
+
+    /// Returns the top 8 elements of the operand stack.
+    fn get_coeff_as_base_elements(&self) -> [Felt; 8] {
+        let c0 = self.stack.get(0);
+        let c1 = self.stack.get(1);
+        let c2 = self.stack.get(2);
+        let c3 = self.stack.get(3);
+        let c4 = self.stack.get(4);
+        let c5 = self.stack.get(5);
+        let c6 = self.stack.get(6);
+        let c7 = self.stack.get(7);
+
+        [c0, c1, c2, c3, c4, c5, c6, c7]
+    }
+
+    /// Returns the top 8 elements of the operand stack.
+    fn get_coeff_as_quad_ext_elements(&self) -> [QuadFelt; 4] {
+        let c0_0 = self.stack.get(0);
+        let c0_1 = self.stack.get(1);
+        let c1_0 = self.stack.get(2);
+        let c1_1 = self.stack.get(3);
+        let c2_0 = self.stack.get(4);
+        let c2_1 = self.stack.get(5);
+        let c3_0 = self.stack.get(6);
+        let c3_1 = self.stack.get(7);
+
+        [
+            QuadFelt::new(c0_0, c0_1),
+            QuadFelt::new(c1_0, c1_1),
+            QuadFelt::new(c2_0, c2_1),
+            QuadFelt::new(c3_0, c3_1),
+        ]
+    }
+
+    /// Returns randomness.
+    fn get_evaluation_point(&mut self) -> Result<QuadFelt, ExecutionError> {
```

Comment:
I would probably change the comment here because evaluation point does not need to be a random value (e.g., for some use cases, we may be just evaluating a polynomial at non-random values).

Reply:
Missed that in the last feedback round. Fixed now